### ISSUE TITLE
[Question] If huggingface transformer can be directly converted by using builder.py in bert demo

### ISSUE THREAD

**Issue by Slyne** [NVIDIA/TensorRT#1048]
## Description
Currently, I'm having an issue that the the engine output result I get by following bert demo is different from the tensorflow2 model result (https://huggingface.co/bert-base-chinese).
The weights names are different so I changed the weights names in order to make sure the weights can be fit into bert demo.

However, I checked the output layer by layer but it turns out only the embedding(including mask, input, segment) layer result is the same, and the rest of the transformer layers are all different. 

Another thing I find is that even the output of each transformer layers are different, the final results(I'm working on named entity recognition task) are similar (after argmax) but not the same.

Just wanna ask if someone has done similar experiment and can share some experience or point out the mistakes I made.

## Environment

**TensorRT Version**:  TensorRT-7.2.1.6
**GPU Type**:  V100
**Nvidia Driver Version**:  450.51
**CUDA Version**:  cuda-11
**CUDNN Version**: 
**Operating System + Version**:  Ubuntu-18.04
**Python Version (if applicable)**:  3.6
**TensorFlow Version (if applicable)**:  model trained in tf2.4,  in bert demo docker tensorflow is 1.15
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**:   Built by demo suggested

## Steps To Reproduce

You may follow the BERT DEMO.
The weights can be downloaded from https://huggingface.co/bert-base-chinese.

**Comment by pommedeterresautee** [comment]
Hi, I have the same issue. I tried to import an onnx model (Pytorch based) converted from Huggingface in Tensorrt using master branch and instructions from `TensorRT/demo/BERT/` but it didn't work.

**Comment by Slyne** [comment]
@pommedeterresautee   I think my issue is not the same as yours.  You may first try to convert the related weights names. For example, the 'cls_squad_output_weights' is only available in this demo not in the tensorflow1 bert ckpt, so you need to convert the weight or remove this weight name.
I can successully convert models trained based on tensorflow1 (tf2 not work) bert by adding or deleting some weights in the checkpoints.   Just wondering if hugging face transformer can be directly applied here ?

**Comment by pommedeterresautee** [comment]
Tks, I have not found any resources making a link between things availables through Huggingface and tensorrt.
Did you manually mapped each layer name for the chinese Bert?

**Comment by pommedeterresautee** [comment]
I don't know if you saw this video, last question at the end, does hugging face support tensorrt, they did the work for their clients but won't share it, open source version optimizes code readability vs performance.


https://events.nvidia.com/meettheexperts5?ncid=so-twit-46587-vt04

**Comment by Slyne** [comment]
> Tks, I have not found any resources making a link between things availables through Huggingface and tensorrt. Did you manually mapped each layer name for the chinese Bert?
Yes.

**Comment by Slyne** [comment]
> @Slyne Could you try TRT 8.2/8.4 and see if the issue still exists? If it does, we will debug it. Thanks

Hi Pohanh, 
I no longer work on this and have moved all my pipeline to pytorch checkpoints. Things seem pretty good.

**Comment by nvpohanh** [comment]
Sounds good. I will close this. Thanks
